### ISSUE TITLE
Pagination of `<ReferenceManyField>`

### ISSUE THREAD

**Issue by leesei** [marmelab/admin-on-rest#561]
`<ReferenceManyField>` is hardcoded to render page 1 of the result.
Any suggestion as to how to extend it to support pagination?

**Comment by fzaninotto** [comment]
We could add a "next" button at the bottom.

But my experience is that users prefer a short list showing an extract of the related content, plus a button to go to the pre-filtered list, where they can use the whole filtering/sorting/pagination.

So for the time being, I'll say that it's out of scope.

**Comment by leesei** [comment]
I see, I'll update the docs to add this tip.

**Comment by kfern** [comment]
Hi @leesei 
Is there an example on how to add a "next" button at the bottom?

**Comment by leesei** [comment]
I ended up using a link to the resource with pre-filterted list rather than add pagination to `<ReferenceManyField>`.

**Comment by MROFerreiro** [comment]
If you want to fetch all records, not having pagination, and have an "infinite" scroll you can set `<ReferenceManyField>` perPage prop to 0. It works with loopback dataProvider, you'll need to test on other dataProviders. Range becomes from 0 to -1 and loopback returns all records.

**Comment by afilp** [comment]
@MROFerreiro I have a resource that has almost 9000 entries, this appropach would not be nice to the server, neither to the UI. Do you think we could have an example with such pagination? It would be helpful. Thanks!

**Comment by MROFerreiro** [comment]
I think that pagination is the best option. I just noticed that fetch all data was possible, so I commented to help people that might need it as a quick solution for their problems.

As I recall, I think `<ReferenceManyField>` calls dataProvider with page hardcoded to 1, as @leesei said. Adding a component to `<ReferenceManyField>` that would render pagination in the UI and change the page variable for dataProvider could be a solution. `<ReferenceManyField>` should also have a flag prop to render this component or not in order to still be viable using it inside `<Datagrid>`. This flag should default to false.

The problem I see with this implementation is make `<ReferenceManyField>` less generic and more propitious to incorrect usage.
